src/string-util/string-util.ts

```diff
@@ -15,50 +15,84 @@ const KOREA_LOT_NUMBER_ADDRESS_REGEXP = /읍|면|동|가/;
 const KOREA_ADDRESS_SORT_REGEXP = new RegExp(
   `${KOREA_ROAD_NAME_ADDRESS_REGEXP.source}|${KOREA_LOT_NUMBER_ADDRESS_REGEXP.source}`
 );
-const EMOJI_REGEXP =
-  /(\u00a9|\u00ae|[\u2000-\u3300]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])/gi;
+
+//이모지 정규식 표현
+const EMOJI_REGEXP = /([\uD800-\uDBFF][\uDC00-\uDFFF])/;
+//ZERO WIDTH JOINER = 다중이모지에서 이모지 끼리 붙일 때 쓰이는 코드
+const EMOJI_ZERO_WIDTH_JOINER = '\u200D';
 
 export namespace StringUtil {
   export function maskPrivacy(text: string, type: PrivacyType): string {
-    const targetMask: Record<PrivacyType, MaskingRange> = {
-      name: {
-        start: EMOJI_REGEXP.test(text.slice(0, 2)) ? 2 : 1,
-        end: EMOJI_REGEXP.test(text.slice(text.length - 2)) ? text.length - 2 : text.length - 1,
-      },
-      phone: {
-        start: /\D/.test(text) ? getMaskingIndexOfUnnormalizedPhone(text).start : 3,
-        end: /\D/.test(text) ? getMaskingIndexOfUnnormalizedPhone(text).end : text.length - 4,
-      },
+    const maskingRule: Record<PrivacyType, MaskingRange> = {
+      name: { start: 1, end: text.length - 1 },
+      phone: { start: 3, end: text.length - 4 },
       email: { start: 2, end: text.indexOf('@') },
       bankAccount: { start: 3, end: text.length - 3 },
       address: { start: getMaskingStartIndexOfAddress(text), end: text.length },
     };
 
-    const start = targetMask[type].start;
-    const end = targetMask[type].end;
+    const start = maskingRule[type].start;
+    const end = maskingRule[type].end;
     const length = type === 'name' ? end - start || 1 : end - start;
     return getMaskedString({ text, length, maskingStart: start });
 
     function getMaskedString({ text, length, maskingStart }: MaskingOpts): string {
+      //emoji가 있는 name에 대해서 따로 처리 = 현재 이모지는 name에만 존재함.
+      if (EMOJI_REGEXP.test(text)) {
+        return getMaskedWithEmoji(text);
+      }
+
       const validLength = Math.max(0, length);
       const masking = '*'.repeat(validLength);
       const textSplit = [...text];
       textSplit.splice(maskingStart, validLength, masking);
+
       return textSplit.join('');
     }
 
-    function getMaskingIndexOfUnnormalizedPhone(phone: string): { start: number; end: number } {
-      const phoneNumberParts = phone.split(/\D/);
+    function getMaskedWithEmoji(text: string): string {
+      const charOfText = enumerateCharOfText(text);
+
+      let maskedName = '';
 
-      let middleIndex = Math.floor(phoneNumberParts.length / 2);
+      charOfText.forEach((char, index) => {
+        if (index !== 0 && (charOfText.length === 2 || index !== charOfText.length - 1)) {
+          maskedName += '*';
+        } else {
+          maskedName += char;
+        }
+      });
+
+      return maskedName;
+    }
+
+    function enumerateCharOfText(text: string): string[] {
+      //emoji 정규식으로 emoji와 emoji가 아닌 글자로 split
+      const splitNameWithEmoji = text.split(EMOJI_REGEXP);
+      const unMaskedName = [];
```

Comment:
그냥 result라고 하는게 맘에 안들면 charList
masking과 상관없는 코드인데 굳이 unmasked 이런 표현을 쓸 이유가 없습니다.
또한 result에 string 이외의 타입이 들어오지 않도록 타입 선언을 하는게 좋겠습니다
```suggestion
      const result: string[] = [];
```

Reply:
결과에 타입 선언 알려주셔서 감사합니다.